goal_src/jak2/engine/gfx/mood/mood.gc

```diff
@@ -5,16 +5,990 @@
+           )
+          (else
+            (set! (-> gp-0 fade) 0.0)
+            (set! (-> gp-0 time) 0.0)
+            )
+          )
+        )
+      (when (not (paused?))
+        (let ((f0-15 (-> gp-0 time)))
+          (set! arg7 (cond
+                       ((= (-> *display* bg-clock clock-ratio) 0.0)
+                        (empty)
+                        arg7
+                        )
+                       (else
+                         (* arg7 (-> pp clock time-adjust-ratio))
+                         )
+                       )
+                )
+          (let ((f0-16 (+ f0-15 arg7)))
+            (set! (-> gp-0 time) f0-16)
+            f0-16
+            )
+          )
+        )
+      )
+    )
+  )
+
+(deftype lava-state (structure)
+  ((lava float  :offset-assert   0)
+   )
+  :method-count-assert 9
+  :size-assert         #x4
+  :flag-assert         #x900000004
+  )
+
+
+(defun update-mood-lava ((arg0 mood-context) (arg1 int) (arg2 int) (arg3 float) (arg4 float) (arg5 float) (arg6 float) (arg7 float))
+  "Unused"
+  (let ((gp-0 (the-as lava-state (+ (+ arg2 1840) (the-as int arg0)))))
+    (let ((f0-1 (cos (-> gp-0 lava))))
+      (set! (-> arg0 times arg1 w) (+ arg3 (* f0-1 arg4)))
+      (set! (-> arg0 times (+ arg1 1) w) (+ arg3 (* (- f0-1) arg4)))
+      )
+    (when (not (paused?))
+      (let ((f0-6 (+ (-> gp-0 lava) arg5)))
+        (set! (-> gp-0 lava) f0-6)
+        f0-6
+        )
+      )
+    )
+  )
+
+(deftype flicker-state (structure)
+  ((flicker-off uint8  :offset-assert   0)
+   (flicker-on  uint8  :offset-assert   1)
+   )
+  :method-count-assert 9
+  :size-assert         #x2
+  :flag-assert         #x900000002
+  )
+
+
+;; WARN: Return type mismatch int vs none.
+(defun update-mood-flicker ((arg0 mood-context) (arg1 int) (arg2 int))
+  "Unused"
+  (let ((gp-0 (the-as flicker-state (+ (+ arg2 1840) (the-as int arg0)))))
+    (cond
+      ((nonzero? (-> gp-0 flicker-on))
+       (set! (-> arg0 times arg1 w) 1.0)
+       (if (not (paused?))
+           (+! (-> gp-0 flicker-on) -1)
+           )
+       )
+      ((nonzero? (-> gp-0 flicker-off))
+       (if (not (paused?))
+           (+! (-> gp-0 flicker-off) -1)
+           )
+       )
+      (else
+        (set! (-> gp-0 flicker-on) (the-as uint (the int (rand-vu-float-range 2.0 20.0))))
+        (if (zero? (the int (rand-vu-float-range 0.0 3.0)))
+            (set! (-> gp-0 flicker-off) (the-as uint (the int (rand-vu-float-range 2.0 120.0))))
+            (set! (-> gp-0 flicker-off) (the-as uint (the int (rand-vu-float-range 2.0 20.0))))
+            )
+        )
+      )
+    )
+  (none)
+  )
+
+(deftype florescent-state (structure)
+  ((value  float  :offset-assert   0)
+   (delay  int8   :offset-assert   4)
+   (delay2 int8   :offset-assert   5)
+   )
+  :method-count-assert 9
+  :size-assert         #x6
+  :flag-assert         #x900000006
+  )
+
+
+(defun update-mood-florescent ((arg0 mood-context) (arg1 int) (arg2 int))
+  "Unused"
```

Comment:
I think it's possible some of these get referenced by name in actual level files, so they might not show up in any `.gc` files, but still get called. But this is just speculation.